src/ensembl/production/metadata/updater/core.py

```diff
@@ -54,40 +54,84 @@ def _load_meta_dict(self):
-                self.meta_dict[species_id][meta_key] = meta_value
+                if meta_key not in self.meta_dict[species_id]:
+                    self.meta_dict[species_id][meta_key] = []
+                self.meta_dict[species_id][meta_key].append(meta_value)
 
     def _validate_required_attributes(self):
         """Check if all required attributes are present in the meta_dict for each species."""
-        required_attribute_names = []
+        # TODO: Move to datacheck
         with self.metadata_db.session_scope() as session:
-            # Query the attribute table to get all required attributes
             required_attributes = session.query(Attribute.name).filter(Attribute.required == 1).all()
             required_attribute_names = {attr.name for attr in required_attributes}
 
-        with self.db.session_scope() as session:
-            # Check each species_id in meta_dict
-            missing_attributes = {}
-            for species_id, meta in self.meta_dict.items():
-                missing = required_attribute_names - set(meta.keys())
-                if missing:
-                    missing_attributes[species_id] = missing
-
-            if missing_attributes:
-                exceptions.MissingMetaException(
-                    "Species ID {species_id} is missing required attributes: {missing_attributes}")
-
-    # Basic API for the meta table in the submission database.
+        missing_attributes = {}
+        for species_id, meta in self.meta_dict.items():
+            if species_id is None:
+                continue
+            missing = required_attribute_names - set(meta.keys())
+            if missing:
+                missing_attributes[species_id] = missing
+
+        if missing_attributes:
+            error_msg = "\n".join([
+                f"Species ID {species_id} is missing required attributes: {', '.join(sorted(missing))}"
+                for species_id, missing in missing_attributes.items()
+            ])
+            raise exceptions.MissingMetaException(error_msg)
+
     def get_meta_single_meta_key(self, species_id, parameter):
+        """
+        Get a single value for a meta_key.
+        Raises an exception if multiple values exist for the same key.
+
+        Returns:
+            str or None: The meta value, or None if not found
+
+        Raises:
+            DuplicateMetaKeyException: If multiple values exist for the key
+        """
         species_meta = self.meta_dict.get(species_id)
         if species_meta is None:
             return None
-        return species_meta.get(parameter)
+
+        values = species_meta.get(parameter)
+        if values is None:
+            return None
+
+        if len(values) > 1:
+            raise exceptions.MetaException(
+                f"Species {species_id} has {len(values)} values for meta_key '{parameter}': {values}. "
+                f"A single key is currently required to successfully hand over."
+            )
```

Comment:
Not sure how likely it is, but if `values` contains a string rather than a list, the exception will be raised saying that the `species_id` has as many values as the length of the string 😅